### PR TITLE
Policy cmdlets implementation and bug fixes

### PR DESCRIPTION
**Pull request by sudhakar4171**

### CODE REVIEW

File: src/ResourceManager/RecoveryServices.Backup/Cmdlets/ProtectionPolicy/PolicyCmdletHelpers.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            {
+                throw new ArgumentException(string.Format(
+                                   "PolicyName length exception - should be within {0} and {1}",
+                                    PolicyConstants.MinPolicyNameLength,
+                                    PolicyConstants.MaxPolicyNameLength));                
+            }
+
+            if (!rgx.IsMatch(policyName))
+            {
+                var exception = new ArgumentException(
```
Comment (sudhakar4171) [inline src/ResourceManager/RecoveryServices.Backup/Cmdlets/ProtectionPolicy/PolicyCmdletHelpers.cs]: will add all exception messages in resources file in one shot and get them reviewed.

File: src/ResourceManager/RecoveryServices.Backup/Cmdlets/ProtectionPolicy/NewAzureRmRecoveryServicesPolicy.cs [hunk 2]
```diff
@@ -55,6 +55,17 @@ public override void ExecuteCmdlet()
...
 
+            // validate policy name
+            PolicyCmdletHelpers.ValidateProtectionPolicyName(Name);
+
+            // Validate if policy already exists
+            string rgName = "";  // TBD
+            string resourceName = "";  // TBD
+            if(PolicyCmdletHelpers.GetProtectionPolicyByName(Name, HydraAdapter, rgName, resourceName) != null)
+            {
+                throw new ArgumentException("Policy already exists with this name:" + Name);
```
Comment (dragonfly91) [inline src/ResourceManager/RecoveryServices.Backup/Cmdlets/ProtectionPolicy/NewAzureRmRecoveryServicesPolicy.cs]: Please use Resources

File: src/ResourceManager/RecoveryServices.Backup/Cmdlets/ProtectionPolicy/GetAzureRmRecoveryServicesPolicy.cs [hunk 3]
```diff
@@ -49,15 +51,28 @@ public class GetAzureRmRecoveryServicesProtectionPolicy : RecoveryServicesBackup
 
         public override void ExecuteCmdlet()
         {
-            // TBD section below
-
             base.ExecuteCmdlet();
 
+            string rgName = "";  // TBD
+            string resourceName = "";  // TBD
+            List<AzureRmRecoveryServicesPolicyBase> respList = new List<AzureRmRecoveryServicesPolicyBase>();
```
Comment (pragrawa) [inline src/ResourceManager/RecoveryServices.Backup/Cmdlets/ProtectionPolicy/GetAzureRmRecoveryServicesPolicy.cs]: We should remove passing of ResourceName and ResourceGroupName from cmdlet as we are not making it optional param, hydra adapter should take these values from vault context.

File: src/ResourceManager/RecoveryServices.Backup/Cmdlets/ProtectionPolicy/SetAzureRmRecoveryServicesPolicy.cs [hunk 4]
```diff
@@ -47,6 +47,19 @@ public override void ExecuteCmdlet()
         {
             base.ExecuteCmdlet();
 
+            // Validate policy name
+            PolicyCmdletHelpers.ValidateProtectionPolicyName(Policy.Name);
```
Comment (pragrawa) [inline src/ResourceManager/RecoveryServices.Backup/Cmdlets/ProtectionPolicy/SetAzureRmRecoveryServicesPolicy.cs]: No need to validate Name as we are not allowing to modify Name

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Providers/Providers/IaasVmPsBackupProvider.cs [hunk 5]
```diff
@@ -60,22 +66,165 @@ public RecoveryPointResponse GetRecoveryPoint()
...
+            string resourceName = (string)providerData.ProviderParameters[PolicyParams.ResourceName];
+            string resourceGroupName = (string)providerData.ProviderParameters[PolicyParams.ResourceGroupName];
+
+
+            // do validations
+            ValidateAzureVMWorkloadType(workloadType);
+
+            // validate both RetentionPolicy and SchedulePolicy
+            ValidateAzureVMRetentionPolicy(retentionPolicy);
+            ValidateAzureVMSchedulePolicy(schedulePolicy);
```
Comment (pragrawa) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Providers/Providers/IaasVmPsBackupProvider.cs]: Add validation for weekly backup schedule, i.e., only weekly retentions are allowed for weekly policy.

**Pull request merged**